plugins/package-managers/node/src/main/kotlin/Yarn.kt

```diff
@@ -88,12 +93,27 @@ class Yarn(
 
         val process = run(workingDir, "info", "--json", packageName)
 
-        val data = Json.parseToJsonElement(process.stdout).jsonObject["data"]?.also {
+        val data = parseYarnInfo(process.stdout)?.jsonObject?.get("data")?.also {
             yarnInfoCache.write(packageName, it.toString())
-        } ?: checkNotNull(Json.parseToJsonElement(process.stderr).jsonObject["data"]).also {
+        } ?: checkNotNull(parseYarnInfo(process.stderr)?.jsonObject?.get("data")).also {
             logger.warn { "Error running '${process.commandLine}' in directory $workingDir: $it" }
         }
 
         return parsePackageJson(data)
     }
 }
+
+/**
+ * Parse the given [output] of a Yarn _info_ command to a [JsonElement] that can be further processed. The output is
+ * typically a JSON object with the metadata of the package that was queried. However, under certain circumstances,
+ * Yarn may return multiple JSON objects separated by newlines; for instance, if the operation is retried due to
+ * network problems. This function filters for the object with the data based on the _type_ field. Result is *null* if
+ * no matching object is found or the input is not valid JSON.
+ */
+internal fun parseYarnInfo(output: String): JsonElement? =
```

Comment:
just question, not sure if idea is good: If the return type was `PackageJson?`, then it would be
more easy to write the test. (e.g. without the `parseAndCheckJson()`, but with a visible assertion in each test
instead.

Reply:
I assume this is possible, but this means that the interaction with the cache would also have to be changed. The cache is populated with the JSON representation of the package metadata. If `parseYarnInfo` returned a `PackageJson`, this object would have to be serialized again. I can try this, or do you think that this has other disadvantages?

Reply:
I have implemented the proposal; `parseYarnInfo` now returns a `PackageJson`.

Reply:
Please have another look @fviernau.